app/components/restaurant/restaurant.controller.js

```diff
@@ -0,0 +1,327 @@
+(function () {
+    'use strict';
+    angular.module('FoodOrderingApp')
+        .controller('RestaurantController', RestaurantController);
+    RestaurantController.$inject = ['$sessionStorage', '$state', '$stateParams', '$uibModal', '$log', '$scope', 'FoodService', 'RestaurantService', 'OrderService'];
+
+    function RestaurantController($sessionStorage, $state, $stateParams, $uibModal, $log, $scope, FoodService, RestaurantService, OrderService) {
+
+        var vm = this;
+        vm.foodItems = [];
+        vm.restaurant = '';
+        vm.foods = [];
+        vm.order = OrderService.getOrder();
+        vm.message = '';
+
+        init();
+
+        function init() {
+            if (angular.isUndefined($sessionStorage.emailId) || $sessionStorage.emailId === '') {
+                $state.go('login');
+            }
+
+            if ($sessionStorage.addFoods) {
+                vm.addFoods = $sessionStorage.addFoods;
+            }
+            else {
+                vm.addFoods = [];
+            }
+
+            $scope.$on('updateFoodList', function (event, data) {
+                vm.foods = data;
+            });
+
+            vm.role = $sessionStorage.role;
+            if ($stateParams.restaurant) {
+                $sessionStorage.restaurant = $stateParams.restaurant;
+                vm.restaurant = $sessionStorage.restaurant;
+                vm.status = vm.restaurant.active;
+            }
+
+            else {
+                vm.restaurant = $sessionStorage.restaurant;
+                vm.status = vm.restaurant.active;
+            }
+        }
+
+        //Getting Foods for the current Restaurant
+        vm.getFood = function () {
+            if (vm.restaurant) {
+                FoodService.getFoodList(vm.restaurant.id)
+                    .then(
+                        function (answer) {
+                            vm.foods = answer.data;
+                        },
+                        function (error) {
+                            $log.info(error);
+                        },
+                        function (progress) {
+                            $log.info(progress);
+                        }
+                    );
+            }
+        };
+
+        vm.getFood();
+
```

Comment:
if getFood() function need to be invoked in controller initialization process then keep it inside init() function